projects/driver_controls/inc/throttle.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+// Module for the throttle.
+// Requires Ads1015, and soft timer to be initialized.
+// The module periodically reads the pedal inputs from ADS1015, translates them into
+// positions and raises the events that correspond to the pedal's position (braking, coasting, and
+// accelerating). It also checks if the readings are stale (in case of a disconnection for
+// example) and could raise a timeout event.
+// To use the module, init the ADS1015 for the pedal and pass its Ads1015Storage to throttle_init,
+// along with the two channels the pedal is connected to. Also pass a ThrottleCalibrationData
+// initialized with desired values. At any time calling throttle_get_position will give the current
+// position of the pedal.
+// Note that storage, pedal_ads1015_storage, and calibration_data should persist.
+
+#include <stdint.h>
+#include "ads1015.h"
+#include "soft_timer.h"
+#include "status.h"
+
+// The time period between every update of the pedal readings.
+#define THROTTLE_UPDATE_PERIOD_MS 50
+
+typedef enum {
+  THROTTLE_ZONE_BRAKE = 0,
+  THROTTLE_ZONE_COAST,
+  THROTTLE_ZONE_ACCEL,
+  NUM_THROTTLE_ZONES
+} ThrottleZone;
+
+typedef enum {
+  THROTTLE_CHANNEL_MAIN = 0,
+  THROTTLE_CHANNEL_SECONDARY,
+  NUM_THROTTLE_CHANNELS
+} ThrottleChannel;
+
+typedef enum { THROTTLE_THRESH_MIN = 0, THROTTLE_THRESH_MAX, NUM_THROTTLE_THRESHES } ThrottleThresh;
+
+typedef int16_t ThrottleZoneThresh[NUM_THROTTLE_CHANNELS][NUM_THROTTLE_ZONES]
```

Comment:
Does the typedef still have value?

Reply:
You mean if it is necessary to have it? I don't think if it was necessary from the beginning lol

Reply:
It's not really a question of "necessary". Typedefs are just syntactic sugar. It's whether having it adds value. Does it make anything more clear?

Reply:
Hmm, knowing it is an int is more clear even!